src/tox/tox_env/api.py

```diff
@@ -303,6 +304,13 @@ def _handle_env_tmp_dir(self) -> None:
             ensure_empty_dir(env_tmp_dir)
         env_tmp_dir.mkdir(parents=True, exist_ok=True)
 
+    def _handle_core_tmp_dir(self) -> None:
+        core_tmp_dir = self.core["temp_dir"]
+        if core_tmp_dir.exists() and next(core_tmp_dir.iterdir(), None) is not None:
+            LOGGER.debug("clear tox temp folder %s", core_tmp_dir)
+            ensure_empty_dir(core_tmp_dir)
```

Comment:
I'm pretty sure this folder is not cleared in tox 3, so likely shouldn't do it here either?

Reply:
We should double-check my statement with the code from the legacy branch. And we should test this functionality as part of the test we're adding 😊

Reply:
yes, using tox3 it does not ensure if temp_dir is empty